File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {

        if(x < 0){
            return false ;
        }

        long revNum = 0 ; 
        long temp = x ; 
        while(temp != 0){
            long lastDigit = temp % 10 ;
            revNum  = (revNum * 10) + lastDigit ; 
            temp /= 10 ;
        }
        return revNum == x ; 
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false ; 
        long rev = 0 ; 
        long temp = x ; 

        while(temp != 0){
            long last = temp % 10 ; 
            rev = (rev*10)+last ; 
            temp /= 10 ; 
        }

        if(rev == x) return true ; 
        return false ; 
    }
}